components/cli/pkg/commands/image/test.go

```diff
@@ -254,196 +266,98 @@ func startTestCellInstance(imageDir string, instanceName string, runningNode *de
+			err = CreateBallerinaConf(string(iName), verboseMode, mntImageDir, string(dependencyLinksJson), envVars, testsRoot)
+			if err != nil {
+				return fmt.Errorf("error occurred while creating the ballerina.conf, %s", err)
 			}
-		}()
-		stderrReader, _ := cmd.StderrPipe()
-		stderrScanner := bufio.NewScanner(stderrReader)
-		go func() {
-			for stderrScanner.Scan() {
-				fmt.Printf("\r\x1b[2K\033[36m%s\033[m\n", stderrScanner.Text())
+			err = RunDockerTelepresenceTests(incell, cmd, testsRoot, imageDir, instanceName)
+			if err != nil {
+				StopTelepresence(telepresenceYamlPath)
+				return fmt.Errorf("error occured while running tests in ballerina docker image, %s", err)
 			}
-		}()
-
-		if exePath != "" {
-			ballerinaArgs := []string{exePath}
-			ballerinaArgs = append(ballerinaArgs, cmdArgs...)
-			RunTelepresenceTests(incell, cmd, ballerinaArgs, imageDir, instanceName, debug)
 		} else {
-			isExistsSourceBal, err := util.FileExists(filepath.Join(balModule, filepath.Base(tempTestFileName)))
-			if !isExistsSourceBal {
-				fileCopyError := util.CopyFile(tempTestFileName, filepath.Join(balModule, filepath.Base(tempTestFileName)))
-				if fileCopyError != nil {
-					util.ExitWithErrorMessage("Error occurred while copying temp source bal file", err)
-				}
-			}
-
-			iNameMarshalled, _ := json.Marshal(string(iName))
-			dependencyLinksMarshalled, _ := json.Marshal(string(dependencyLinksJson))
-			ballerinaArgs := []string{
-				fmt.Sprintf("%s", string(iNameMarshalled)),
-				fmt.Sprintf("%s", string(dependencyLinksMarshalled)),
-				startDependenciesFlag,
-				shareDependenciesFlag}
-
-			var scriptCmds []string
-			scriptCmds = append(scriptCmds, []string{constants.DockerCliBallerinaExecutablePath}...)
-			scriptCmds = append(scriptCmds, cmdArgs...)
-			scriptCmds = append(scriptCmds, ballerinaArgs...)
-
-			dockerCommand := []string{"/bin/bash", "-c", fmt.Sprintf("'%s'", strings.Join(scriptCmds, " "))}
-			dockerCmdArgs := SetupBallerinaDocker(string(iName), string(dependencyLinksJson), verboseMode, imageDir, testsRoot, scriptCmds)
-			dockerCmdArgs = append(dockerCmdArgs, dockerCommand...)
-
-			RunDockerTelepresenceTests(incell, cmd, dockerCmdArgs, imageDir, instanceName, debug)
-		}
-
-		_ = os.Remove(tempTestFileName)
-		if err != nil {
-			return fmt.Errorf("failed to execute test method in Cell instance %s due to %v", instanceName, err)
-		}
-	} else {
-		isExistsSourceBal, err := util.FileExists(filepath.Join(balModule, filepath.Base(balFilePath)))
-		if !isExistsSourceBal {
-			fileCopyError := util.CopyFile(balFilePath, filepath.Join(balModule, filepath.Base(balFilePath)))
-			if fileCopyError != nil {
-				util.ExitWithErrorMessage("Error occurred while copying source bal file", err)
+			err = CreateBallerinaConf(string(iName), verboseMode, imageDir, string(dependencyLinksJson), envVars, testsRoot)
+			if err != nil {
+				return fmt.Errorf("Error occurred while creating the ballerina.conf, %s", err)
```

Comment:
suggestion:
return fmt.Errorf("error occurred while creating the ballerina.conf, %s", err)

Reply:
FIxed